codebase-pipelines/buildspec-manifests.yml

```diff
@@ -14,17 +14,28 @@ phases:
   build:
     commands:
       - set -e
-      - for env in $(echo $ENVIRONMENTS | jq -c -r '.[]');
+      - |
+        for env in $(echo $ENVIRONMENTS | jq -c -r '.[].name');
         do
-          EXPORT_ENV=$(echo $env | tr '[:lower:]' '[:upper:]');
-          export CLUSTER_NAME_$EXPORT_ENV="$APPLICATION-$env";
+          unset AWS_ACCESS_KEY_ID
+          unset AWS_SECRET_ACCESS_KEY
+          unset AWS_SESSION_TOKEN
+          ACCOUNT_ID=$(echo $ENVIRONMENTS | jq -c -r '.[] | select(.name=='\"$env\"').account.id')
+          assumed_role=$(aws sts assume-role --role-arn "arn:aws:iam::$ACCOUNT_ID:role/$APPLICATION-$env-codebase-pipeline-deploy" --role-session-name "$env-codebase-pipeline-deploy")
+          export AWS_ACCESS_KEY_ID=$(echo $assumed_role | jq -r .Credentials.AccessKeyId)
+          export AWS_SECRET_ACCESS_KEY=$(echo $assumed_role | jq -r .Credentials.SecretAccessKey)
+          export AWS_SESSION_TOKEN=$(echo $assumed_role | jq -r .Credentials.SessionToken)
+          EXPORT_ENV=$(echo $env | tr '[:lower:]' '[:upper:]')
+          export CLUSTER_NAME_$EXPORT_ENV="$APPLICATION-$env"
           for svc in $(echo $SERVICES | jq -c -r '.[]');
           do
-            echo '[{"name":"'$svc'","imageUri":"'$REPOSITORY_URL':'$IMAGE_TAG'"}]' > image-definitions-$svc.json;
-            SERVICE_NAME=$(aws ecs list-services --cluster $APPLICATION-$env | jq -r '.serviceArns[] | select(contains("'$APPLICATION-$env'-'$svc'-Service"))');       
-            EXPORT_SVC="$(echo $EXPORT_ENV'_'$svc | tr - _ | tr '[:lower:]' '[:upper:]')";
-            export SERVICE_NAME_$EXPORT_SVC=$(echo $SERVICE_NAME | cut -d '/' -f3);
-            cat image-definitions-$svc.json;
+            if [ ! -f "image-definitions-$svc.json" ]; then
+              echo '[{"name":"'$svc'","imageUri":"'$REPOSITORY_URL':'$IMAGE_TAG'"}]' > image-definitions-$svc.json
+            fi
+            SERVICE_NAME=$(aws ecs list-services --cluster $APPLICATION-$env | jq -r '.serviceArns[] | select(contains("'$APPLICATION-$env'-'$svc'-Service"))')
+            EXPORT_SVC="$(echo $EXPORT_ENV'_'$svc | tr - _ | tr '[:lower:]' '[:upper:]')"
+            export SERVICE_NAME_$EXPORT_SVC=$(echo $SERVICE_NAME | cut -d '/' -f3)
```

Comment:
Possibly worth a comment or two here to explain what's being transformed into what and why

Reply:
I generally prefer shell scripts to wrap variables in curly braces. e.g .${SERVICE_NAME}. Makes it clearer what the var name is and prevents errors like if someone removes a hyphen.

Reply:
Yeah good point I'll add some comments.

I agree, but unfortunately because of the Terraform templating of this file to inject the exported variables at the top, it can't handle the ${} syntax. It would try and replace it with a Terraform value.

Reply:
Oh OK. 😬